dpnp/dpnp_iface_manipulation.py

```diff
@@ -906,6 +906,14 @@ def asfarray(a, dtype=None, *, device=None, usm_type=None, sycl_queue=None):
 
     """
 
+    warnings.warn(
```

Comment:
I think it might be helpful to create a lingering doc page like CuPy has: `docs/source/reference/_deprecated.rst`. It would list all the active deprecations.
So the functions from that page will keep its docstring rendered, but not indexed.